tests/integration/test_benchmark.py

```diff
@@ -112,3 +115,82 @@ def test_duplicate_synthesizers():
             synthesizers=['GaussianCopulaSynthesizer', 'GaussianCopulaSynthesizer'],
             custom_synthesizers=[custom_synthesizer, custom_synthesizer]
         )
+
+
+def test_benchmark_single_table():
+    """Test all synthesizers, as well as some generated ones, against a dataset.
+
+    The custom synthesizers should be generated from both ``create_single_table_synthesizer``
+    and ``create_sdv_synthesizer_variant``, to test they work.
+    """
+    # Setup
+    def get_trained_synthesizer(data, metadata):
+        model = GaussianCopulaSynthesizer(metadata)
+        model.fit(data)
+        return model
+
+    def sample_from_synthesizer(synthesizer, n_samples):
+        return synthesizer.sample(n_samples)
+
+    TestSynthesizer = create_single_table_synthesizer(
+        display_name='TestSynthesizer',
+        get_trained_synthesizer_fn=get_trained_synthesizer,
+        sample_from_synthesizer_fn=sample_from_synthesizer
+    )
+
+    CTGANVariant = create_sdv_synthesizer_variant(
+        'CTGANVariant',
+        'CTGANSynthesizer',
+        synthesizer_parameters={'epochs': 100}
+    )
+
+    FastMLVariant = create_sdv_synthesizer_variant(
+        'FastMLVariant',
+        'FastMLPreset',
+        synthesizer_parameters={'name': 'FAST_ML'}
+    )
+
+    # Run
+    results = sdgym.benchmark_single_table(
+        synthesizers=[
+            'TVAESynthesizer',
+            'CopulaGANSynthesizer',
+            'GaussianCopulaSynthesizer',
+            'FastMLPreset',
+            'DataIdentity',
+            'IndependentSynthesizer',
+            'UniformSynthesizer',
+            'CTGANSynthesizer'
+        ],
+        custom_synthesizers=[FastMLVariant, TestSynthesizer, CTGANVariant],
+        sdv_datasets=['student_placements']
+    )
+
+    # Assert
+    expected_synthesizers = pd.Series([
+        'TVAESynthesizer',
+        'CopulaGANSynthesizer',
+        'GaussianCopulaSynthesizer',
+        'FastMLPreset',
+        'DataIdentity',
+        'IndependentSynthesizer',
+        'UniformSynthesizer',
+        'CTGANSynthesizer',
+        'Variant:FastMLVariant',
+        'Custom:TestSynthesizer',
+        'Variant:CTGANVariant'
+    ], name='Synthesizer')
+    pd.testing.assert_series_equal(results['Synthesizer'], expected_synthesizers)
+
+    assert set(results['Dataset']) == {'student_placements'}
+    assert set(results['Dataset_Size_MB']) == {0.026358}
+    assert results['Train_Time'].between(0, 1000).all()
+    assert results['Peak_Memory_MB'].between(0, 100).all()
+    assert results['Synthesizer_Size_MB'].between(0, 100).all()
+    assert results['Sample_Time'].between(0, 100).all()
+    assert results['Evaluate_Time'].between(0, 100).all()
+    assert results['Quality_Score'].between(.6, 1).all()
+    assert results['NewRowSynthesis'][4] == 0
+
+    results['NewRowSynthesis'][4] = 1
```

Comment:
Just out of curiosity, why is the 4th value different?

Reply:
Because all values generated by the IdentitySynthesizer match the original data, so there are no new rows.

Reply:
In that case, minor nitpick but could we make a variable to store `results['NewRowSynthesis'][4]` as like `identity_new_row_synthesis_value` or something so it's clear what's being evaluated here?